tests/integration/test_base_service.py

```diff
@@ -99,8 +113,11 @@ def test_base_service_process_request_scenario1(dispatcher, request):
     response_set_config = dispatcher.process_request(request_set_config)
     assert response_set_config.status_code == 201
 
-    response_service_1 = dispatcher.process_request(request_service_1)
-    assert response_service_1.body == b'test_scenario1/TestServiceA/0'
+    with patch('requests.request'):
+        response_service_1 = dispatcher.process_request(request_service_1)
+        assert response_service_1.body == b'test_scenario1/TestServiceA/0'
+
+        sleep(1)  # Wait for the async call to finish before deregistering the mock
```

Comment:
shouldn't you check here if the callback was made? Maybe it's time to add a waiting loop.